src/server/wesnothd/server.cpp

```diff
@@ -1743,16 +1743,21 @@ void server::handle_player_in_game(player_iterator p, simple_wml::document& data
 					}
 				}
 
-				// add leader data
+				// determine leader(s) for the side
```

Comment:
```suggestion
				// approximately determine leader(s) for the side like the client does
```
IMO the key point is to make clear that we already know this is not accurate. So 1) there is no expectation that this is 100% accurate info (unlike pretty much all the other data we save in the DB) and 2) that it's obvious this can be improved if someone wants to since it is not accurate yet (and maybe never will be...).

Reply:
You really think so? What happens with `[leader] canrecruit=no` as a trivial example? Or a unit with invalid coordinates? Or multiple units with the same coordinates?